From 7643d860e4a78fcc58717a09cb8d2198e4367c64 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 29 Mar 2023 18:49:49 +0200
Subject: [PATCH 0368/1141] soc: st: extend STM32 HDP support to stm32mp25

The HDP (Hardware Debug Port) is present on stm32mp25.

Extend driver's dependency to allow compiling it for stm32mp25.
Simplify the default statement; no reason to repeat the same
condition used for 'depend'.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Id9c990016e4013e6607bae4e7f78c7900e48650a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/313881
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/soc/st/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/st/Kconfig b/drivers/soc/st/Kconfig
index 9a970096d29b..6d9e85261706 100644
--- a/drivers/soc/st/Kconfig
+++ b/drivers/soc/st/Kconfig
@@ -1,9 +1,9 @@
 if ARCH_STM32
 
 config STM32_HDP
-	bool "STMicroelectronics STM32MP157 Hardware Debug Port (HDP) pin control"
-	depends on MACH_STM32MP157 || MACH_STM32MP13
-	default n if MACH_STM32MP157 || MACH_STM32MP13
+	bool "STMicroelectronics STM32 Hardware Debug Port (HDP) pin control"
+	depends on MACH_STM32MP157 || MACH_STM32MP13 || MACH_STM32MP25 || ARM64
+	default n
 	help
 	  The Hardware Debug Port allows the observation of internal signals. By using multiplexers,
 	  up to 16 signals for each of 8-bit output can be observed.
-- 
2.39.2

